internal/.kratos.yaml

```diff
@@ -98,15 +104,37 @@ selfservice:
           hooks:
             -
               hook: revoke_active_sessions
+
     registration:
       ui_url: http://test.kratos.ory.sh/register
       lifespan: 98m
       after:
         default_browser_return_url: https://self-service/registration/return_to
         password:
           hooks:
-            -
-              hook: session
+            - hook: session
+            - hook: web-hook
+              config:
+                url: https://test.kratos.ory.sh/after_registration_hook
+                method: POST
+                auth:
+                  type: api-key
+                  config:
+                    name: My-API-Key
+                    value: secret
+                    in: header
+        global:
```

Comment:
How about:

```
    registration:
       ui_url: http://test.kratos.ory.sh/register
       lifespan: 98m
       after:
         default_browser_return_url: https://self-service/registration/return_to
         hooks:
           - hook: session
         password:
           hooks: # overrides `hooks` (does NOT merge!)
             - hook: session
             - hook: web-hook
```

Reply:
Can do this. This will however require some more changes as the
```.go
func HookStrategyKey(key, strategy string) string {
	return fmt.Sprintf("%s.%s.hooks", key, strategy)
}
```
cannot be used as is anymore for "global" hooks

Reply:
> And sorry for being a bit late on this one...
Thank you for taking your time for review! :)

Reply:
> ... plus we will need some documentation for this :)

Definitely. I would update the documentation as soon as all configuration details are clarified and you consider the implementation as complete in sense of this PR

Reply:
> ... so maybe just pass a few things like header, url, method

Will add it to the next update to this PR.

Reply:
```.yaml
password:
           hooks: # overrides `hooks` (does NOT merge!)
```

Are you sure, you don't want to merge? This would mean config duplication. If a hook should run always, one would have to define it for e.g. `password` and `oidc`

Reply:
On the other hand overriding would solve a particular question, which was open: In which sequence should the hooks be executed?
I mean one could define
```.yaml
registration:
       ui_url: http://test.kratos.ory.sh/register
       lifespan: 98m
       after:
         default_browser_return_url: https://self-service/registration/return_to
         hooks: <-------- defined before strategy
           - hook: session
         password:
           hooks: # overrides `hooks` (does NOT merge!)
             - hook: session
             - hook: web-hook
```

as well as 

```.yaml
registration:
       ui_url: http://test.kratos.ory.sh/register
       lifespan: 98m
       after:
         default_browser_return_url: https://self-service/registration/return_to
         password:
           hooks: # overrides `hooks` (does NOT merge!)
             - hook: session
             - hook: web-hook
         hooks: <-------- defined after strategy
           - hook: session
```
If the hooks are merged, the sequence is not defined, if a strategy always overrides a global hook, the sequence is clear.

Reply:
> Are you sure, you don't want to merge? This would mean config duplication. If a hook should run always, one would have to define it for e.g. password and oidc.
>
> I think the current implementation is more user friendly as there is no additional semantic: global hooks run always, strategy specific hooks run only for the particular strategy.

So the problem is the order in which things are executed can't be changed. For example:

```
       after:
         default_browser_return_url: https://self-service/registration/return_to
         password:
           hooks:
             - hook: a
             - hook: b
         hooks:
           - hook: c
           - hook: d
```

If we merge these, then what is being executed in what order?

- a, b, c, d
- c, d, a, b
- a, c, b, d
- what if you want: b, d, a, c?

This is important because some hooks might cancel the request and thus order is important. By not merging these keys, but making this explicit, you have to write a bit more config but at least the result is predictable. I am strongly in favor of not merging these configs.